tarantool/src/main/java/com/yahoo/ycsb/db/TarantoolClient.java

```diff
@@ -64,12 +62,12 @@ public Status insert(String table, String key, Map<String, ByteIterator> values)
     return replace(key, values, "Can't insert element");
   }
 
-  private HashMap<String, ByteIterator> tupleConvertFilter(List<String> input, Set<String> fields) {
-    HashMap<String, ByteIterator> result = new HashMap<>();
+  private static <T extends Map<String, ByteIterator>>
+      T tupleConvertFilter(List<String> input, Set<String> fields, T result) {
```

Comment:
why introduce an out parameter here?

Reply:
To reuse this method in two cases:
- read single row where resulting map is given (and it is Map - we cannot accept HashMap as parameter)
- read a row from scan where we need to create our own map and add it to the result (and it must be HashMap - we cannot return Map)

Alternatively this method can be void with usage 
```
      HashMap<String, ByteIterator> temp = new HashMap<String, ByteIterator>();
      tupleConvertFilter(i, fields, temp);
```
but it still need to have `<T extends Map<String, ByteIterator>>` parameter.